### ISSUE TITLE
Error 500

### ISSUE THREAD

**Issue by B-codes** [mevdschee/php-crud-api#265]
I've completed the DB section and checked prerequisites. I keep coming up with Error 500.

Is there a known reason why php-crud-api won't run on shared hosting?

**Comment by Denis-Evseev** [comment]
Error 500 - Internal server error.
=> Check logs

**Comment by B-codes** [comment]
I did this and found this:

[06-Jun-2017 19:18:54 UTC] PHP Fatal error:  Uncaught Exception: Connect
failed. Access denied for user 'ats_gpf'@'localhost' (using password:
YES) in /home/ats/.../api.php:115
Stack trace:
#0 /home/atservic/.../api.php(2187): MySQL->connect('localhost',
'ats_gpf', 'password', 'ats_gpf1', NULL, NULL, 'UTF8')
#1 /home/atservic/.../api.php(2697): PHP_CRUD_API->__construct(Array)
#2 {main}
  thrown in /home/atservic/.../api.php on line 115

Odd thing is that the password is correct. Everything has been checked
and triple checked on my end. Am I missing a PHP module dependency or
something?

I have checked Mysqli is installed. And other apps are functioning on
this server and able to connect to Mysql

**Comment by B-codes** [comment]
So I've solved error 500 by recompiling Apache with default settings. 

I'm still getting the above error. 

Am I missing any prerequisites that are not listed on your Readme page?
Anyone else have any joy on shared hosting?

**Comment by mevdschee** [comment]
Hi, sorry for your bad experience. It seems that there is some mysql permission issue. Can you try to execute this example?


    <?php

    $link = mysqli_init();
    if (!$link) {
        die('mysqli_init failed');
    }

    if (!mysqli_options($link, MYSQLI_INIT_COMMAND, 'SET AUTOCOMMIT = 0')) {
        die('Setting MYSQLI_INIT_COMMAND failed');
    }

    if (!mysqli_options($link, MYSQLI_OPT_CONNECT_TIMEOUT, 5)) {
        die('Setting MYSQLI_OPT_CONNECT_TIMEOUT failed');
    }

    if (!mysqli_real_connect($link, 'localhost', 'my_user', 'my_password', 'my_db')) {
        die('Connect Error (' . mysqli_connect_errno() . ') '
                . mysqli_connect_error());
    }

    echo 'Success... ' . mysqli_get_host_info($link) . "\n";

    mysqli_close($link);
    ?>


[source](http://php.net/manual/en/mysqli.real-connect.php)

**Comment by B-codes** [comment]
Thanks for your reply.

I've pasted the connection info from a password manager in both cases to
eliminate typo's

It returns a success prompt.
Success... Localhost via UNIX socket

**Comment by B-codes** [comment]
But oddly I'm unable to get the same result using the API code

**Comment by mevdschee** [comment]
Could it have to do with single or double quoted strings and character escaping in the password?

**Comment by B-codes** [comment]
I'm using single quotes as in the original example. I've changed
passwords several times and only use an exclamation as a special character.

What I find odd is that your test script runs, but as soon as I run the
API I get no data back. It only outputs a numeric 1 in the source of the
output page.

I've oddly stopped getting connection failed errors in the error.log but
the response has not changed.
If it will help I can PM you specifics like the public URL we're using
and the Database config section and sample table data we're testing with.

The API runs fine on a local Xampp install though.

**Comment by B-codes** [comment]
I found the solution. Been scratching our heads and recompiled our entire Apache stack. Your suggestion around escape characters gave me some ideas. I simplified the special characters in the password somewhat. Then the access denied issue dissapeared in the logs. But I still had no data coming back.

Untill I reviewed the code again and found I had forgotten to uncomment this line:
$api->executeCommand();

So the API did not execute after login to return data.
